stdlib/LinearAlgebra/src/diagonal.jl

```diff
@@ -102,6 +101,7 @@ julia> Diagonal(A)
 """
 Diagonal(A::AbstractMatrix) = Diagonal(diag(A))
 Diagonal{T}(A::AbstractMatrix) where T = Diagonal{T}(diag(A))
+Diagonal{T,V}(M::AbstractMatrix) where {T,V<:AbstractVector{T}} = Diagonal{T,V}(diag(M))
```

Comment:
Just for consistency with surrounding code:

```suggestion
Diagonal{T,V}(A::AbstractMatrix) where {T,V<:AbstractVector{T}} = Diagonal{T,V}(diag(A))
```